centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1821,9 +1840,9 @@ objDataCompare(int data_type_1, void* data_ptr_1, int data_type_2, void* data_pt
 		switch(data_type_2)
 		    {
 		    case DATA_T_MONEY: 
-			if (m->WholePart > ((pMoneyType)data_ptr_2)->WholePart) cmp_value = 1;
-			else if (m->WholePart < ((pMoneyType)data_ptr_2)->WholePart) cmp_value = -1;
-			else cmp_value = m->FractionPart - ((pMoneyType)data_ptr_2)->FractionPart;
+			if (m->MoneyValue/10000 > ((pMoneyType)data_ptr_2)->MoneyValue/10000) cmp_value = 1;
+			else if (m->MoneyValue/10000 < ((pMoneyType)data_ptr_2)->MoneyValue/10000) cmp_value = -1;
+			else cmp_value = m->MoneyValue%10000 - ((pMoneyType)data_ptr_2)->MoneyValue%10000;
```

Comment:
same as above, you don't need to do divide or modulus, and just set cmp_value to 0 in the last else